diff --git a/node_modules/onnxruntime-react-native/android/CMakeLists.txt b/node_modules/onnxruntime-react-native/android/CMakeLists.txt
index 2f814e8..191ae18 100644
--- a/node_modules/onnxruntime-react-native/android/CMakeLists.txt
+++ b/node_modules/onnxruntime-react-native/android/CMakeLists.txt
@@ -97,3 +97,9 @@ target_link_libraries(
   ${log-lib} # <-- Logcat logger
   android # <-- Android JNI core
 )
+
+target_link_options(
+  onnxruntimejsi PRIVATE
+  "-Wl,-z,max-page-size=16384"
+  "-Wl,-z,common-page-size=16384"
+)
diff --git a/node_modules/onnxruntime-react-native/android/build.gradle b/node_modules/onnxruntime-react-native/android/build.gradle
index 41b4359..81c429c 100644
--- a/node_modules/onnxruntime-react-native/android/build.gradle
+++ b/node_modules/onnxruntime-react-native/android/build.gradle
@@ -62,7 +62,7 @@ def readPackageJsonField(field) {
   }
 }
 
-boolean ortExtensionsEnabled = readPackageJsonField('onnxruntimeExtensionsEnabled') == "true"
+boolean ortExtensionsEnabled = readPackageJsonField('onnxruntimeExtensionsEnabled') == "false"
 boolean useQnn = readPackageJsonField('onnxruntimeUseQnn') == "true"
 
 def REACT_NATIVE_VERSION = ['node', '--print', "JSON.parse(require('fs').readFileSync(require.resolve('react-native/package.json'), 'utf-8')).version"].execute(null, rootDir).text.trim()
@@ -84,7 +84,9 @@ android {
         cppFlags "-O2 -frtti -fexceptions -Wall -Wno-unused-variable -fstack-protector-all"
         if (REACT_NATIVE_MINOR_VERSION >= 71) {
           // fabricjni required c++_shared
-          arguments "-DANDROID_STL=c++_shared",
+          arguments 
+            "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON",
+            "-DANDROID_STL=c++_shared",
             "-DNODE_MODULES_DIR=${nodeModules}",
             "-DORT_EXTENSIONS_ENABLED=${ortExtensionsEnabled}",
             "-DREACT_NATIVE_VERSION=${REACT_NATIVE_VERSION}",
@@ -247,7 +249,7 @@ dependencies {
     extractLibs "com.microsoft.onnxruntime:onnxruntime-android:latest.integration@aar"
   }
 
-  if (VersionNumber.parse(REACT_NATIVE_VERSION) < VersionNumber.parse("0.71")) {
+  if (REACT_NATIVE_VERSION.split("\\.")[0].toInteger() == 0 && REACT_NATIVE_VERSION.split("\\.")[1].toInteger() < 71) {
     extractLibs "com.facebook.fbjni:fbjni:+:headers"
     extractLibs "com.facebook.fbjni:fbjni:+"
   }
